### PR TITLE
[NOARCH] Add a recipe for helios-menuinst

### PR DESCRIPTION
**Pull request by dokempf**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by dokempf** [comment]
@conda-forge/help-python This is ready for review. Sorry for hijacking the Python team here, there is no "Generic" team to ping for review.

Maybe a note on what this package does is in order: I am in the process of creating `constructor`-based installers for the package `helios`. For that I would like to add `menuinst` menus, but restrict them to `constructor` installers and ignore them in regular terminal installations. I consulted jaimergp on the constructor Matrix chat and he confirmed that putting the `menuinst` JSONs into a separate package is the way to go.

File: recipes/helios-menuinst/run_test.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+if ! test -f $PREFIX/Menu/helios.json; then
```
Comment (jaimergp) [inline recipes/helios-menuinst/run_test.sh:3]: You can probably get by by adding this directly in meta.yaml as:

```
test -f $PREFIX/Menu/helios.json
```

File: recipes/helios-menuinst/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+source:
+  path: src
+
+build:
+  noarch: generic
+  number: 0
+  script:
+    - mkdir -p {{ PREFIX }}/Menu
+    - cp {{ SRC_DIR }}/menu.json {{ PREFIX }}/Menu/helios.json
```
Comment (jaimergp) [inline recipes/helios-menuinst/meta.yaml:13]: The target json file must match or start with the package name.

```suggestion
    - cp {{ SRC_DIR }}/menu.json {{ PREFIX }}/Menu/helios-menuinst.json
```

File: recipes/helios-menuinst/run_test.sh (hunk 1 above)
Comment (dokempf) [reply recipes/helios-menuinst/run_test.sh:3]: I did something like that and it executed with `sh` instead of `bash` and produced an error.

Comment (jaimergp) [reply recipes/helios-menuinst/run_test.sh:3]: Hm, weird. That's standard posix 🤔 Can you try again so we can see the traceback?

**Comment by jaimergp** [comment]
The important bit for menuinst is the conda-build validation (which doesn't error out if it fails, only a warning; to be addressed in the future).

In this case, the logs do say:

```
Validating Menu/*.json files
'Menu/helios.json' is a valid menuinst JSON document
```

So we are fine.

Note: given how the build script is Unix-y, Windows builds are failing, but given how simple the JSON document is, it should work ok.

**Comment by dokempf** [comment]
@jaimergp It does indeed pass now with the tests in `meta.yaml`. I am not sure what exactly I had done wrong before (and force-pushed it away).

**Review by jaimergp — APPROVED** [review]

**Pull request merged**